gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/ExecutionTreeBuilder.java

```diff
@@ -53,6 +54,18 @@ public QueryNode build() {
     // All expressions' attributes from the same source. Will only need one downstream query.
     Optional<String> singleSourceForAllAttributes = ExecutionTreeUtils.getSingleSourceForAllAttributes(executionContext);
 
+    EntitiesRequest entitiesRequest = executionContext.getEntitiesRequest();
+
+    // TODO: If there is a filter on a data source, other than EDS, then the flag is a no-op
+    if (!entitiesRequest.getQueryLiveEntitiesOnly()) {
+      QueryNode rootNode =
+          new DataFetcherNode(EDS.name(), entitiesRequest.getFilter());
+      executionContext.removePendingSelectionSource(EDS.name());
```

Comment:
rootNode.acceptVisitor(new ExecutionContextBuilderVisitor(executionContext));
 instead of the explicit removePending* calls